sandbox/libs/authn/src/main/java/org/opensearch/authn/DefaultObjectMapper.java

```diff
@@ -43,11 +52,172 @@ public static void inject(final InjectableValues.Std injectableValues) {
+    }
+
+    public static <T> T readTree(JsonNode node, Class<T> clazz) throws IOException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
+                @Override
+                public T run() throws Exception {
+                    return objectMapper.treeToValue(node, clazz);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
     public static <T> T readValue(String string, Class<T> clazz) throws IOException {
-        return objectMapper.readValue(string, clazz);
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
+                @Override
+                public T run() throws Exception {
+                    return objectMapper.readValue(string, clazz);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
+    public static JsonNode readTree(String string) throws IOException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<JsonNode>() {
+                @Override
+                public JsonNode run() throws Exception {
+                    return objectMapper.readTree(string);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
     }
 
     public static String writeValueAsString(Object value, boolean omitDefaults) throws JsonProcessingException {
-        return (omitDefaults ? defaulOmittingObjectMapper : objectMapper).writeValueAsString(value);
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<String>() {
+                @Override
+                public String run() throws Exception {
+                    return (omitDefaults ? defaulOmittingObjectMapper : objectMapper).writeValueAsString(value);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (JsonProcessingException) e.getCause();
+        }
+
+    }
+
+    public static <T> T readValue(String string, TypeReference<T> tr) throws IOException {
```

Comment:
For my knowledge: why do we need this readValue and one one line 189, and in which scenarios will they be used?

Reply:
`public static <T> T readValue(String string, Class<T> clazz)` is used to read from the `internal_users.yml` and map to User objects. All other readValue that weren't being used were removed.